restapi.py

```diff
@@ -0,0 +1,85 @@
+import logging
+from threading import Thread
+
+from werkzeug.wrappers import Request, Response
+from werkzeug.routing import Map, Rule
+from werkzeug.exceptions import BadRequest, MethodNotAllowed, HTTPException
+
+log = logging.getLogger("rest")
+
+
+class BotRestApi(object):
+    def __init__(self, messageFun):
+        """Initialize the REST API
+        :param messageFun: callable, a function that will be invoked when a message was sent via the web-ui"""
+        self.messageFun = messageFun
+        self.form = open("res/static/chatform.html", "rb").read()
+        self.url_map = Map(
+            [Rule("/", endpoint="root"), Rule("/message", endpoint="message")]
+        )
+
+    def dispatch_request(self, request):
+        adapter = self.url_map.bind_to_environ(request.environ)
+        try:
+            endpoint, values = adapter.match()
+            return getattr(self, "on_" + endpoint)(request, **values)
```

Comment:
I don't know how you feel about this, but I don't really like to use this approach and call methods dynamically. Maybe it's better to define a mapping instead?

Reply:
This was taken from the Werkzeug tutorial, I didn't think about it that much, assuming that whatever they have in their guide is a best practice. At the moment we don't have many endpoints and features and I suppose that it will probably stay that way - this is just an internal use API that is not exposed anywhere, used manually by the admin. The risks are minimal.